db/db_impl.cc

```diff
@@ -3109,74 +3109,110 @@ Status DBImpl::IngestExternalFile(
     ColumnFamilyHandle* column_family,
     const std::vector<std::string>& external_files,
     const IngestExternalFileOptions& ingestion_options) {
-  if (external_files.empty()) {
-    return Status::InvalidArgument("external_files is empty");
-  }
-
-  Status status;
-  auto cfh = reinterpret_cast<ColumnFamilyHandleImpl*>(column_family);
-  auto cfd = cfh->cfd();
+  return IngestExternalFiles({column_family}, {external_files},
+                             {ingestion_options});
+}
 
-  // Ingest should immediately fail if ingest_behind is requested,
-  // but the DB doesn't support it.
-  if (ingestion_options.ingest_behind) {
-    if (!immutable_db_options_.allow_ingest_behind) {
+Status DBImpl::IngestExternalFiles(
+    const std::vector<ColumnFamilyHandle*>& column_families,
+    const std::vector<std::vector<std::string>>& external_files,
+    const std::vector<IngestExternalFileOptions>& ingestion_options_list) {
```

Comment:
Do you still plan to make it a vector of a struct?

Reply:
Actually I have concerns for both choices.
**Vector of ingestion options**. If we use a vector of ingestion options, is it possible that we update `IngestExternalFileOptions` and it becomes impossible to ingest files for multiple column families atomically in the future?
**Single ingestion options**. Some users may put SST files of different column families in different places. Is it possible that users would like to *copy* SST for some column families, but *move* SST for others?